core/test/matrix/dense.cpp

```diff
@@ -135,6 +135,7 @@ TYPED_TEST(Dense, CanBeConstructedFromExistingData)
 
     ASSERT_EQ(m->get_const_values(), data);
     ASSERT_EQ(m->at(2, 1), value_type{6.0});
+    ASSERT_EQ(m->at(2, 2), value_type{-1.0});
```

Comment:
Honestly I don't think we should be testing for this. If a developer implements something that violates this somehow, I don't think that should be counted as a bug. I agree that we should not be modifying data that falls outside this matrix, so m[2,2] should still be -1. But I'd rather check it as follows:
```suggestion
    ASSERT_EQ(m->get_const_values()[2*m->get_stride() + 2], value_type{-1.0});
```
That is, the underlying array should be unchanged were the view is not supposed to touch it, but we don't guarantee access to it through the view. That said, I'm okay with whatever you decide here.

Reply:
For this case, isn't it that we are just making sure that the underlying data is not modified ? Both the ways, check for the same thing. For me, the second one is a bit more complex to understand, so I would prefer to keep it the simpler way.